### PR TITLE
Fix some type checks on the signatures of nested step handlers

### PR DESCRIPTION
**Pull request by Johnlon**
### 🤔 What's changed?

This change fixes some type checks on the signatures and runtime return values of the handlers. 
This change also improves the quality of the error messages produced.

--

History ...

At some point someone changed the return type for "nested steps" handlers from []string to godog.Steps but they forgot to adjust the type checks where the handlers are registered.

They got away with this because the existing type checks were lax and unable to distinguish  []string from godog.Steps.
However,  in a couple of places in the code the value is coerced to godog.Steps and so if someone had actually returned []string then the code would blow up at run time rather than during registration. 

Additionally there were some tests around these types but they also had not been updated and the test was passing for the wrong reason - the particular test expected an error but the cause of the error was an accident caused by the coercion mentioned above rather than a delibarate act of the test.

### ⚡️ What's your motivation? 

Bug fix and better error handling.

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, etc. without changing behaviour)
- :bug: Bug fix (non-breaking change which fixes a defect)

### ♻️ Anything particular you want feedback on?

I'm pretty sure the change is good - comments please.

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://github.com/cucumber/.github/tree/main?tab=coc-ov-file)
- [ x] I've changed the behaviour of the code
  - [x ] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Comment by Johnlon** [comment]
@mpkorstanje @vearutop - mind taking a look please.

The PR description explains it all.
Pretty sure this is all good but wouldn't mind some other eyes.

File: internal/formatters/fmt_output_test.go [hunk 1]
```diff
@@ -179,9 +179,6 @@ func fmtOutputTest(fmtName, testName, featureFilePath string) func(*testing.T) {
 		expected := normalise(string(expectedOutput))
 		actual := normalise(buf.String())
 		assert.Equalf(t, expected, actual, "path: %s", expectOutputPath)
-		if expected != actual {
```
Comment (Johnlon) [inline internal/formatters/fmt_output_test.go:182]: I removed some junk

**Comment by mpkorstanje** [comment]
The PR looks reasonable but I don't have enough familiarity to review Go.

**Review by vearutop — APPROVED** [review]
LGTM, thanks!

**Comment by Johnlon** [comment]
thanks folk. I have a bunk of other wip I need finish.

**Pull request merged**